run_cert.py

```diff
@@ -0,0 +1,333 @@
+    else:
+        return ""
+
+
+def write_domains_to_file(domains):
+    """Store list of domains in file to retrieve on next run"""
+    with open(DOMAINS_FILE, "w+") as domains_file:
+        domains_file.write(domains)
+
+
+def find_newest_dir(domain_name):
+    dirs = {}
+    # Check if the certificate has been recreated with a new domain list
+    for x in os.listdir(CERTIFICATES_DIR):
+        if x.startswith(domain_name):
+            # Dict all the directories and their creation times in the CERTIFICATES_DIR that start with the domain_name
+            xpath = f"{CERTIFICATES_DIR}/{x}"
+            dirs[xpath] = os.path.getctime(xpath)
+    # Return the newest directory in the dict
+    list_dir = sorted(dirs.items(), key=operator.itemgetter(1))
+    return list_dir[-1][0]
+
+
+def write_combined_cert_to_file(domain_name):
+    """Create the combined cert from the full chain and private key files"""
+    if domain_name.startswith("*"):
+        domain_name = domain_name.replace("*.", "")
+
+    latest_cert_dir = find_newest_dir(domain_name)
+
+    full_chain_file = f"{latest_cert_dir}/fullchain.pem"
+    private_key_file = f"{latest_cert_dir}/privkey.pem"
+    combined_cert = f"{latest_cert_dir}/{domain_name}.pem"
+
+    print("Writing combined certificate for marathon-lb to this path: %s" % combined_cert, flush=True)
+    with open(full_chain_file) as f:
+        with open(combined_cert, "w+") as f1:
+            for line in f:
+                f1.write(line)
+
+    # Marathon-lb requires that the entire certificate chain including private key
+    # be combined into a single PEM formatted file.
+    with open(private_key_file) as f:
+        with open(combined_cert, "a") as f1:
+            for line in f:
+                f1.write(line)
+
+
+def configure_provider_creds():
+    """Configure DNS credentials if required"""
+    verification_method = os.environ.get(ENV_VERIFICATION_METHOD, "http")
+    if verification_method == "dns":
+        dns_provider = os.environ.get(ENV_DNS_PROVIDER)
+        if dns_provider == "google":
+            service_account = os.environ.get(ENV_GCE_SERVICE_ACCOUNT, "")
+            if len(service_account) > 0:
+                with open(DEFAULT_GCE_CREDENTIALS, "w+") as creds_file:
+                    creds_file.write(service_account)
+
+                os.chown(DEFAULT_GCE_CREDENTIALS, 0, 0)
+                os.chmod(DEFAULT_GCE_CREDENTIALS, stat.S_IREAD)
+                print("Creating GCE Service Account file", flush=True)
+            else:
+                raise Exception("GCE_SERVICE_ACCOUNT is not defined")
+        elif dns_provider == "route53":
+            if len(os.environ.get("AWS_ACCESS_KEY_ID", "")) == 0 or len(
+                    os.environ.get("AWS_SECRET_ACCESS_KEY", "")) == 0:
+                raise Exception("AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY is not defined")
+        else:
+            raise Exception("Unknown DNS verification method")
```

Comment:
seems like this error should read "Unknown DNS provider"
   and the enclosing if condition should have something like 
```   
   elif verification_method !="http"
        raise Exception("Unknown DNS verification method")
```